.github/workflows/build.yml

```
@@ -30,15 +30,11 @@ name: Build
 
 on: [push, pull_request]
 
-jobs:
+concurrency:
+  group: ${{ github.workflow }}-${{ (github.ref == 'refs/heads/main' && github.run_id) || github.event.pull_request.number || github.ref }}
```

Is it possible one day we have some release branches and also want this behavior of main branch on those branches?
```suggestion
  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.run_id }}
```

---

I think we can deal with it once we have release branches in future. resolve it now.

---

Different branches should not cancel each other.
I think github.run_id already works as intended.

What issue have we not deal with?

---

say we created a release branch `stable`, we may want runs on the `stable` branch won't be cancelled just like `main`.

---

I think `${{ github.workflow }}-${{ github.event.pull_request.number || github.run_id }}` is simpler and works better, too.

---

I'd also like to cancel previously running jobs when pushing to a development branch in my fork.

---

I'm fine with the current rule. And maybe we can add additional `github.repository` check.
```suggestion
  group: ${{ github.workflow }}-${{ github.event.pull_request.number || (github.repository == 'openthread/openthread' && github.run_id) || github.ref }}
```